pytorch_lightning/plugins/precision/native_amp.py

```diff
@@ -106,6 +105,15 @@ def pre_optimizer_step(
             self.scaler.update()
         return False
 
+    def post_backward(
+        self, model: "pl.LightningModule", closure_loss: torch.Tensor, optimizer: torch.optim.Optimizer
+    ) -> torch.Tensor:
+        ret_val = super().post_backward(model, closure_loss, optimizer)
+        # unscale here to have it inside the closure before the grad tracking and clipping
+        if not model.trainer.fit_loop._should_accumulate():
+            self.scaler.unscale_(optimizer)
```

Comment:
Can you check this isn't the LightningOptimizer ?

Reply:
done, added assertion